var/spack/repos/builtin/packages/genomefinisher/package.py

```diff
@@ -35,7 +35,7 @@ def install(self, spec, prefix):
 
         # Munge the helper script to explicitly point to java and the jar file
         # jar file.
-        java = spec['jdk'].prefix.bin.java
+        java = spec['java'].prefix.bin.java
```

Comment:
This change seems legit, but maybe it should be better for any `java` provider to set this variable in the dependent module scope?

Reply:
@adamjstewart :point_up:

Reply:
I still haven't figured out the best way to handle Java packages.